src/configure/templateInputHelper/InputControl.ts

```diff
@@ -62,21 +73,31 @@ export class InputControl {
         this.visible = value;
     }
 
-    public async setInputControlValue(azureSession: AzureSession): Promise<any> {
+    public async setInputControlValue(): Promise<any> {
         if (!this.isVisible()) {
             return;
         }
         if (!!this.dataSource) {
             var dependentInputs = this._getDataSourceInputs();
             if (this.controlType === ControlType.None || this.controlType === ControlType.InputBox) {
-                this.value = await this.dataSource.evaluateDataSources(dependentInputs, azureSession);
+                this.value = await this.dataSource.evaluateDataSources(dependentInputs, this.azureSession);
```

Comment:
Won't we need to trigger dynamic/static validation for input box?

Reply:
ahh okay we are only getting the value here

Reply:
What if the value returned by datasource is invalid? 
We should still run validation and if validation fails then show the input to user. What do you think? Otherwise user will get an error and they will not even know hot fix it.

Reply:
yes that's right, but user may not know the correct value for it. But he should have the option to fix the value. Added it.